lib/trivial.nix

```diff
@@ -13,6 +13,19 @@ rec {
     # The value to return
     x: x;
 
+  /* imports a flake.nix without acknowledging its lock file, useful for
+     referencing subflakes from a parent flake. The second argument allows
+     specifying the inputs of this flake.
+
+     Example:
+       callLocklessFlake ./directoryContainingFlake { inherit nixpkgs; }
+  */
+  callLocklessFlake = path: inputs: let
+    r = {outPath = path;} //
```

Comment:
```suggestion
    self = {outPath = path;} //
```

Reply:
> `{outPath = path;}`

Canonicalize `outPath`? Depending on the type, might need something like `builtins.path { name = "source": ... }`.

Not sure, maybe this should be the caller's responsibility. If they want to avoid putting intermediate paths in the store, perhaps we shouldn't.

Reply:
In most cases I'd see this used via a sub-flake in a monorepo or another flake input which would already be canonicalized.

Reply:
I agree. Let's not make the mistake again of doing too much.

Reply:
I don't have a strong opinion either way at the moment.

I'm not sure what the behavior is with builtins.path in term of multiple copies. I'm thinking of cases like a monorepo where a subdirectory already exists in the Store somewhere, just not as a top-level store path.

Reply:
After some thought, this scenario seems rather serious:

1. user sees that `outPath` isn't copied, uses it for some sources imports, etc
2. user applies `callLocklessFlake` to directory with confidential files, which is safe to do
3. we change `callLocklessFlake` to match other users' expectations of calling `builtins.path`
4. user updates Nixpkgs
5. user unknowingly puts confidential files in the store or binary cache

A similar scenario exists for the opposite direction, where a store path size (and therefore closure size) can increase unknowingly.

It seems like it's a matter of time before someone comes along to make it do the "right" thing and enact (3), so let's preempt that by adding a `copy` parameter without a default, fixing the problem at (1). We can assign a default once flakes stabilize and when we know how this function is used in practice.

> I'm not sure what the behavior is with builtins.path in term of multiple copies.

Subflakes always call it, so `outPath` is always a top-level path. It may not always do this in the future, but for now it seems that we should replicate that behavior. (Maybe I've been holding it wrong, or maybe this changes)
There's no need for the flake to be a subflake and normal flakes also put `outPath` in the store on way or another.